### PR TITLE
let users pick a updateStrategy for statefulsets since they default i…

### PR DESCRIPTION
**Pull request by grosser**
…s bad

OnDelete might be supported once I get https://github.com/zendesk/samson/pull/2241 working, but it's kinda wonky ... so nudge them into RollingUpdate

@jonmoter @irwaters 

/cc @jacobat  @libo

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/role_verifier.rb [hunk 1]
```diff
@@ -146,11 +147,22 @@ def verify_project_and_role_consistent
...
       return unless service = @elements.detect { |t| t[:kind] == "Service" }
-      return unless set = @elements.detect { |t| t[:kind] == "StatefulSet" }
+      return unless set = find_stateful_set
       return if set.dig(:spec, :serviceName) == service.dig(:metadata, :name)
       @errors << "Service metadata.name and StatefulSet spec.serviceName must be consistent"
     end
 
+    def verify_stateful_set_restart_policy
+      return unless set = find_stateful_set
+      return if set.dig(:spec, :updateStrategy)
```
Comment (libo) [inline plugins/kubernetes/app/models/kubernetes/role_verifier.rb]: I am not sure this matches our yml file structure. https://github.com/zendesk/help-center-memcached-k8s/blob/a0790e62a4cbda6fc52172f1ae79154d6d385679/kubernetes/memcached.yml#L20

Honestly, I rather have @jacobat and @dasch opinion here. Not sure if we use `OnDelete` or `RollingUpdate `

**Comment by dasch** [comment]
This only seems to be supported in k8s 1.7. But yeah, RollingUpdate seems to be the one we want.

File: plugins/kubernetes/app/models/kubernetes/role_verifier.rb (hunk 1 above)
Comment (dasch) [reply plugins/kubernetes/app/models/kubernetes/role_verifier.rb]: It looks :+1: to me.

Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/role_verifier.rb]: the default is OnDelete, that's why I'm calling it out explicitly

**Pull request merged**